### PR TITLE
Some fixes to NLE and the competition task

### PR DESCRIPTION
**Pull request by heiner**
This PR fixes our C++ lint test (which apparently never really worked and at any rate was broken for months now) and flat-out disables seeding in NLE. The seeding commit is rather defensive for purposes of the NLE competition. Afterwards, we can `#define NLE_ALLOW_SEEDING 1` to turn on the old behaviour.

### CODE REVIEW

File: nle/nethack/actions.py [hunk 1]
```diff
@@ -153,7 +153,15 @@ class Command(enum.IntEnum):
...
     SEEALL = ord("*")  # show all equipment in use
+    SEEAMULET = ord('"')  # show the amulet currently worn
+    SEEARMOR = ord("[")  # show the armor currently worn
+    SEEGOLD = ord("$")  # count your gold
+    SEERINGS = ord("=")  # show the ring(s) currently worn
+    SEESPELLS = ord("+")  # list and reorder known spells
+    SEETOOLS = ord("(")  # show the tools currently in use"
     SEETRAP = ord("^")  # show the type of adjacent trap
+    SEEWEAPON = ord(")")  # show the weapon currently wielded
+    SHELL = ord("!")  # do a shell escape (not enabled in NLE build)
```
Comment (condnsdmatters) [inline nle/nethack/actions.py:164]: Can we just leave this out 
```suggestion
```

File: nle/nethack/actions.py [hunk 2]
```diff
@@ -15,11 +15,11 @@ def C(c):
 
 
 class TextCharacters(enum.IntEnum):
-    PLUS = ord("+")
+    PLUS = ord("+")  # Also SEESPELLS.
```
Comment (condnsdmatters) [inline nle/nethack/actions.py:18]: Seems silly to have the same actions twice. Anyone looking to "lookup" a key press will have trouble. Much cleaner to make sure the mapping is one-to-one. I would remove here.

File: nle/env/tasks.py [hunk 3]
```diff
@@ -341,6 +341,13 @@ def __init__(
         # If the in-game turn count doesn't change for 10_000 steps, we abort
         self.no_progress_timeout = no_progress_timeout
 
+        def f(*args, **kwargs):
```
[the remark this answers is not in the document]
Comment (condnsdmatters) [reply nle/env/tasks.py:344]: Ah i have seen below now that the compilation changes result in thrown exceptions in pynethack.cc so this really is window dressing, presumably to be _extra clear_ to competitors?

File: version.txt [hunk 4]
```diff
@@ -1 +1 @@
-0.7.1
+0.7.2
```
Comment (condnsdmatters) [inline version.txt:1]: Would do this in a different PR. we also need to possible change SAVEDIR etc

**Review by condnsdmatters — COMMENTED** [review]
The code all looks fine except I wouldnt mark this for release yet. Some discussion points listed. The only think I'm thinking about is:  do we whole hog overhaul action space or make an incremental change? Another option (crazy) is to provide mapping to old action space in some way?

File: include/config.h [hunk 5]
```diff
@@ -55,6 +55,7 @@
 #define RL_GRAPHICS  /* Reinforcement learning 'window' port */
 
 #define NO_SIGNAL
+#define NOSHELL
```
[the remark this answers is not in the document]
Comment (heiner) [reply include/config.h:58]: Makes it such that https://github.com/facebookresearch/nle/blob/ceee6396797c5fe00eac66aa909af48e5ee8b04d/include/unixconf.h#L316 doesn't define `SHELL` such that https://github.com/facebookresearch/nle/blob/ceee6396797c5fe00eac66aa909af48e5ee8b04d/sys/unix/unixunix.c#L267 doesn't check `sysopt.shellers` for whether the USER CAN EXECUTE A SHELL COMMAND but instead refuses it for all users.

File: nle/env/tasks.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (heiner) [reply nle/env/tasks.py:344]: History of this change: I wanted to remove the Python functions that do seed setting, but unfortunately the pybind11 object's attributes are read-only so they cannot be changed this way.

I guess we could not do this now.

File: nle/nethack/actions.py [hunk 6]
```diff
@@ -112,11 +112,11 @@ class Command(enum.IntEnum):
     DROP = ord("d")  # drop an item
     DROPTYPE = ord("D")  # drop specific item types
     EAT = ord("e")  # eat something
-    ESC = C("[")  # escape from the current query/action
     ENGRAVE = ord("E")  # engrave writing on the floor
     ENHANCE = M("e")  # advance or check weapon and spell skills
-    FIRE = ord("f")  # fire ammunition from quiver
+    ESC = C("[")  # escape from the current query/action
```
[the remark this answers is not in the document]
Comment (heiner) [reply nle/nethack/actions.py:117]: Agreed.

The change here is from sorting the entries, which they were initially and then I inserted `ESC` at slightly the wrong spot.

File: nle/nethack/actions.py (hunk 1 above)
Comment (heiner) [reply nle/nethack/actions.py:164]: If we do that we also need to add `!` to the not-available ASCII characters in the action test this PR adds.

I guess this depends on whether we want to be able to name something with `!`. I guess we don't.

File: nle/nethack/actions.py (hunk 2 above)
Comment (heiner) [reply nle/nethack/actions.py:18]: The lookup dict deals with that. I guess for symmetry reasons I found it nicer to have `PLUS` available given that `MINUS` is. I'm fine with either.

File: version.txt (hunk 4 above)
Comment (heiner) [reply version.txt]: Done.

**Comment by heiner** [comment]
Updated PR to not touch actions for now.

File: nle/nethack/actions.py (hunk 2 above)
Comment (heiner) [reply nle/nethack/actions.py]: (Removed action changes from PR.)

**Review by condnsdmatters — APPROVED** [review]
